src/components/common/Balance.js

```diff
@@ -0,0 +1,38 @@
+import React from 'react'
+
+// denomination of one near in minimal non divisible units (attoNears)
+// const NEAR_NOMINATION = 10 ** 18
+const NOMINATION = 18
+
+const REG = /(?=(\B)(\d{3})+$)/g;
+
+const Balance = (props) => {
+    let style = {
+        width: "1em",
+        height: "1em",
+        marginLeft: "3px",
+        verticalAlign: "middle"
+    }
+
+    let index = props.amount ? props.amount.indexOf(".") : null
+    let amount = index > 0 ?  props.amount.slice(0,index): props.amount
+    let zerosSmall = "0".repeat(NOMINATION - 3)
+    let zeros = "0".repeat(NOMINATION)
+    let amountShow = amount ? ((amount.length <= NOMINATION - 3) ?
```

Comment:
this is somewhat hard to read as a single expression. Maybe extract 2 separate functions: 
1) format as full NEAR tokens
2) format as milliNEAR tokens

It's perfectly ok to have functions which return JSX.

Also this is something that would be good idea to unit test – shouldn't require much setup. See e.g. https://stackoverflow.com/questions/49468126/how-to-test-a-stateless-component